source/common/http/conn_manager_impl.cc

```diff
@@ -58,12 +58,18 @@ ConnectionManagerImpl::ConnectionManagerImpl(ConnectionManagerConfig& config,
                                              Runtime::RandomGenerator& random_generator,
                                              Tracing::HttpTracer& tracer, Runtime::Loader& runtime,
                                              const LocalInfo::LocalInfo& local_info,
-                                             Upstream::ClusterManager& cluster_manager)
+                                             Upstream::ClusterManager& cluster_manager,
+                                             Server::OverloadManager* overload_manager)
     : config_(config), stats_(config_.stats()),
       conn_length_(new Stats::Timespan(stats_.named_.downstream_cx_length_ms_)),
       drain_close_(drain_close), random_generator_(random_generator), tracer_(tracer),
       runtime_(runtime), local_info_(local_info), cluster_manager_(cluster_manager),
-      listener_stats_(config_.listenerStats()) {}
+      listener_stats_(config_.listenerStats()),
+      overload_inactive_(Server::OverloadActionState::Inactive),
```

Comment:
It's a lot odd to me that we need to track a separate variable for this, which is I guess modified by reference by the thread local state? Any reason we can't just query the current state from the thread local state directly? That seems like a better API? Thoughts?

Reply:
That would be my vote FWIW. Which has table lookup? You mean TLS? That should be a vector index.

Reply:
The hash table look up is in the [ThreadLocalOverloadState](https://github.com/envoyproxy/envoy/blob/master/include/envoy/server/overload_manager.h#L32) object which contains a map from overload action string to its state. 

fwiw, @htuch had commented on this issue in my previous [PR](https://github.com/envoyproxy/envoy/pull/4090/files/88f1f911f156c02b5e695eaf51089dc9098de0c1).

Reply:
Yeah, the idea was to avoid string hashing/comparison all the time. It just seemed icky to be performing an unnecessary O(n) string operation (n is the length of the action name) on the data plane fast path (stream creation). I like @eziskind's solution, since it gives good performance and is still quite readable.